test/install.c

```diff
@@ -277,11 +277,11 @@ filename=bootloader.img";
 	g_assert_true(g_hash_table_contains(tgrp, "demofs"));
 	g_assert_true(g_hash_table_contains(tgrp, "bootloader"));
 	g_assert_true(g_hash_table_contains(tgrp, "prebootloader"));
-	//Deactivated check as the actual behavior is GHashTable-implementation-defined
+	//Deactivated checks as the actual behavior is GHashTable-implementation-defined
 	//g_assert_cmpstr(((RaucSlot*)g_hash_table_lookup(tgrp, "rescue"))->name, ==, "rescue.0");
-	g_assert_cmpstr(((RaucSlot*)g_hash_table_lookup(tgrp, "rootfs"))->name, ==, "rootfs.1");
-	g_assert_cmpstr(((RaucSlot*)g_hash_table_lookup(tgrp, "appfs"))->name, ==, "appfs.1");
-	g_assert_cmpstr(((RaucSlot*)g_hash_table_lookup(tgrp, "demofs"))->name, ==, "demofs.1");
```

Comment:
Ok, despite we can determine a valid active slot group here (which is '0'), this obviously fails for the same reason as above, because we have a triple-redunancy setup and nothing that determines whether group '1' or group '2' should be selected.

To not simply drop the tests, I would slightly change the logic so that it at least tests the minimum assertion we need from the algorithm:

``` diff
@@ -279,9 +279,9 @@ filename=bootloader.img";
 	g_assert_true(g_hash_table_contains(tgrp, "prebootloader"));
 	//Deactivated check as the actual behavior is GHashTable-implementation-defined
 	//g_assert_cmpstr(((RaucSlot*)g_hash_table_lookup(tgrp, "rescue"))->name, ==, "rescue.0");
-	g_assert_cmpstr(((RaucSlot*)g_hash_table_lookup(tgrp, "rootfs"))->name, ==, "rootfs.1");
-	g_assert_cmpstr(((RaucSlot*)g_hash_table_lookup(tgrp, "appfs"))->name, ==, "appfs.1");
-	g_assert_cmpstr(((RaucSlot*)g_hash_table_lookup(tgrp, "demofs"))->name, ==, "demofs.1");
+	g_assert_cmpstr(((RaucSlot*)g_hash_table_lookup(tgrp, "rootfs"))->name, !=, "rootfs.0");
+	g_assert_cmpstr(((RaucSlot*)g_hash_table_lookup(tgrp, "appfs"))->name, !=, "appfs.0");
+	g_assert_cmpstr(((RaucSlot*)g_hash_table_lookup(tgrp, "demofs"))->name, !=, "demofs.0");
 	g_assert_cmpstr(((RaucSlot*)g_hash_table_lookup(tgrp, "bootloader"))->name, ==, "bootloader.0");
 	g_assert_cmpstr(((RaucSlot*)g_hash_table_lookup(tgrp, "prebootloader"))->name, ==, "prebootloader.0");
 	g_assert_cmpint(g_hash_table_size(tgrp), ==, 6);
```

Reply:
Can we fix the disabled test with the same approach?

Reply:
@jluebbe Not with this approach. For rescue slots we only have a simple redundancy (`rescue.0` and `rescue.1`). There we cannot check for anything except for having at least on of these chosen (in contrast to none is chosen).

Reply:
@ejoerns: Looks good to me, also my builds are happy with this. Thanks